### PR TITLE
JDK-8293177: Verify version numbers in legal files

### PR DESCRIPTION
**Pull request by hns**
Please review the addition of a new JavaDoc test to check the version of 3rd party libraries in legal files against the versions of actual library code. The skeleton for this test (mostly looking up files in the source tree) was taken from `CheckManPageOptions.java`. The test uses a regular expression to extract the name and version of 3rd party libraries in the `share/legal` directory and then matches these strings against the contents of the actual library files. 

The 3rd party libraries currently in use by JavaDoc are jQuery and jQuery-UI so the test infrastructure has some bits that facilitate JavaScript code such as support for both minified and unprocessed files, but it should work equally well for Java and other code.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293177](https://bugs.openjdk.org/browse/JDK-8293177): Verify version numbers in legal files

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [3d8e7826](https://git.openjdk.org/jdk/pull/11211/files/3d8e78263260cb24f4ea2825bfab485d1fcaf3ae)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11211/head:pull/11211` \
`$ git checkout pull/11211`

Update a local copy of the PR: \
`$ git checkout pull/11211` \
`$ git pull https://git.openjdk.org/jdk pull/11211/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11211`

View PR using the GUI difftool: \
`$ git pr show -t 11211`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11211.diff">https://git.openjdk.org/jdk/pull/11211.diff</a>

</details>

### CODE REVIEW

File: test/langtools/jdk/javadoc/doclet/checkLibraryVersions/CheckLibraryVersions.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    static final Map<String, List<String>> libraries = Map.of(
+            "jquery.md", List.of("jquery-%V%M.js"),
+            "jqueryUI.md", List.of("jquery-ui%M.js", "jquery-ui%M.css")
+    );
+
+    public static void main(String... args) throws Exception {
+        try {
+            new CheckLibraryVersions().run(args);
+        } catch (SourceDirNotFound e) {
+            System.err.println("NOTE: Cannot find src directory; test skipped");
```
Comment (jonathan-gibbons) [inline test/langtools/jdk/javadoc/doclet/checkLibraryVersions/CheckLibraryVersions.java:61]: These days, it is common practice (but not required) to throw `jtreg.SkippedException` to convey this info.

**Review by jonathan-gibbons — APPROVED** [review]
Approved, with an optional comment about code style

File: test/langtools/jdk/javadoc/doclet/checkLibraryVersions/CheckLibraryVersions.java (hunk 1 above)
Comment (hns) [reply test/langtools/jdk/javadoc/doclet/checkLibraryVersions/CheckLibraryVersions.java]: Right, I changed the code to use `jtreg.SkippedException`.

**Pull request closed without merging**